### DISCUSSION TITLE
Batch alignment for IMC

### DISCUSSION THREAD

**Discussion by michaeljpitcher** [ImmuneDynamics/Spectre#31] [! News]
Thanks for a very useful package! I have a query: what's the best way to do batch alignment for Imaging mass cytometry data? I have IMC samples that have varying intensities due to technical variance which needs to be accounted for, but I need to make sure we don't lose the biological variance that's present too. 

I have performed arcsinh transformation, but do I also need to perform the batch alignment as described [here](https://wiki.centenary.org.au/pages/viewpage.action?pageId=157272386)? And if so, do I need to treat each image as a separate 'batch'?

**Comment by tomashhurst** [comment]
Hi @michaeljpitcher , 

Thanks for your feedback! Batch alignment is a bit tricky with IMC data. In theory yes you could do it in a similar way, but there are a number of difficulties for getting this to work properly. 

I do have a few suggestions though -- one option is after the asinh transformation, you can re-scale the data from each batch separately between say 0 and 1 (so the max value from each batch is 1). That will _approximately_ align the cellular data between the batches to an extent. However, this might be an issue if some markers are genuinely expressed at lower levels in the samples in some batches. 

The other approach you could take is rather than clustering the cells to identify populations, you can actually pre-classify them in Ilastik, which is something I am doing more often now.

Do you happen to have some kind of control sample recorded with each batch?

**Reply by michaeljpitcher** [reply]
Thanks for the reply. Yeah, it does seem a complicated issue. The problem with scaling 0 to 1 is as you say, one of our samples is missing a particular cell type due to the location of tissue in which the image was taken, and thus the expression of the marker for that cell type is justifiably low. So scaling would make it appear as though some cells were positive for that marker when we know they're not.
I've been working with ilastik to classify objects and then analyse each cell type separately and I agree that that's probably the best way to go (I wish we did have control samples but it seems not). 
Thanks for your advice and for a really interesting package!
